json-rpc/src/tests/unit_tests.rs

```diff
@@ -183,9 +183,15 @@ fn test_transaction_submission() {
     runtime.spawn(async move {
         let validator = MockVMValidator;
         while let Some((txn, cb)) = mp_events.next().await {
-            let vm_status = validator.validate_transaction(txn).unwrap().status();
-            let result = if vm_status.is_some() {
-                (MempoolStatus::new(MempoolStatusCode::VmError), vm_status)
+            let vm_status = validator
+                .validate_transaction(txn.clone())
+                .unwrap()
+                .status();
+            let result = if let Some(vm_status) = vm_status {
+                (
+                    MempoolStatus::new(MempoolStatusCode::VmError),
+                    Some((txn, vm_status)),
+                )
```

Comment:
well that's not entirely true because now we modify every single API else where... in fact, I would argue that we should /expect/ an equal number of transactions going in as statuses coming out... and if that fails, we likely have a pretty nasty bug. So hardening around potential issues in the future seems ... unintuitive since that would mask programming bugs.

Reply:
good point, I fixed this and it looks much simpler now :) and no unwrap